pypose/module/dynamics.py

```diff
@@ -6,7 +6,587 @@
+
+        One may also generate the system matrices with the time variable :obj:`_t`.
+
+        >>> n_batch, n_state, n_ctrl, T = 2, 4, 3, 5
+        >>> n_sc = n_state + n_ctrl
+        >>> x = torch.randn(n_state)
+        >>> u = torch.randn(T, n_ctrl)
+        ... 
+        >>> class MyLTV(pp.module.LTV):
+        ...     def __init__(self, A, B, C, D, T):
+        ...         super().__init__(A, B, C, D)
+        ...         self.T = T
+        ... 
+        ...     @property
+        ...     def A(self):
+        ...         return torch.eye(4, 4) * self._t.cos()
+        ... 
+        ...     @property
+        ...     def B(self):
+        ...         return torch.eye(4, 3) * self._t.sin()
+        ... 
+        ...     @property
+        ...     def C(self):
+        ...         return torch.eye(4, 4) * self._t.tan()
+        ... 
+        ...     @property
+        ...     def D(self):
+        ...         return torch.eye(4, 3)
+        ... 
+        >>> ltv = MyLTV()
+        >>> for t in range(T):
+        ...     x, y = ltv(x, u[t])
+
+    Note:
+        More practical examples can be found at `examples/module/dynamics
+        <https://github.com/pypose/pypose/tree/main/examples/module/dynamics>`_.
+    '''
+    def __init__(self, A=None, B=None, C=None, D=None, c1=None, c2=None):
+        super().__init__(A, B, C, D, c1, c2)
+        self.register_buffer('_t',torch.tensor(0, dtype=torch.int64))
+        self.register_forward_hook(self.forward_hook)
+
+    def forward_hook(self, module, inputs, outputs):
+        r'''
+        Automatically advances the time step.
+        '''
+        self._t.add_(1)
+
+    def reset(self, t=0):
+        self._t.fill_(t)
+        return self
+
+    @property
+    def systime(self):
+        r'''
+            System time, automatically advanced by :obj:`forward_hook`.
+        '''
+        return self._t
+
+    @systime.setter
+    def systime(self, t):
+        if not isinstance(t, torch.Tensor):
+            t = torch.tensor(t)
+        self._t.copy_(t)
+
+    def set_refpoint(self, state=None, input=None, t=None):
+        r'''
+        Function to set the reference point for linearization.
+
+        Args: 
+            state (:obj:`Tensor`): The reference state of the dynamical system. If ``None``,
+                the the most recent state is taken. Default: ``None``.
+            input (:obj:`Tensor`): The reference input to the dynamical system. If ``None``,
+                the the most recent input is taken. Default: ``None``.
+            t (:obj:`Tensor`): The reference time step of the dynamical system. If ``None``,
+                the the most recent timestamp is taken. Default: ``None``.
+
+        Returns:
+            The ``self`` module.
+
+        Warning:
+            For nonlinear systems, the users have to call this function before getting the
+            linearized system.
+        '''
+        self.systime = t
+        return self
+
+
+class NTI(System):
```

Comment:
@wang-chen I completely don't like the idea of separating the nonlinear into NTI, NTV, people never call these concepts at all in general. Even in your code implementation, there are also not much difference between those two with the only difference is the systime t, which shouldn't be ignored even under your concept of NTI, because for a nonlinear system, time can never be ignored, unlike the LTI you can have some common properties, in nonlinear system, time really matter. User usually want to reset time, re-initialize the x(0), or even split a slice of state at some specific t. 
Also, if this is the case, why are we still leave the set_refpoint in the base class. In short , I still think the previous based class with set_refpoint and systime makes more sense. And as a system, A,B,C,D either from direct user input or jacobian linearization are important properties there.